### PR TITLE
Divides species in subtypes, makes viruses able to infect certain subtypes

### LINKED ISSUE #32832: Black slime toxin doesn't work
If you're not a slimeperson, the disease progresses through 5 states. At stage 3, it turns you into a slimeperson, and then at stage 5 it turns you into a slime simplemob. And when it first starts at stage 1, it checks if you're a slimeperson and if you are instantly skips to stage 5. So, no matter how you use it you have to become a slime person first and then either infect yourself or allow the disease to progress... but there's one problem.

Slime people are immune to diseases.

Yeah.

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: Added two new symptoms: one allows viruses to still work while dead, and allows infection of undead species, and one allows infection of inorganic species (such as plasmapeople or golems).
/:cl:

Species are now separated into Organic, Inorganic, Undead and Robotic. The virusimmune tag was kept for species who are effectively organic but immune to viruses for special reasons (abductors).

You can now set the types of species viruses can infect; i made some adjustments that i thought made sense (beesease infecting zombies, parrot possession infecting anyone).

Added process_dead as an additional effect for the new symptom, allowing the disease to tick on corpses; it doesn't have many uses i can think of, but it can heal up people in case of a defibrillation or something like that.

Fixes #32832

### CODE REVIEW

**Comment by Supermichael777** [comment]
slime-people don't need special treatment, abductors need it for game balance purposes.

File: code/datums/diseases/parrotpossession.dm [hunk 1]
```diff
@@ -11,6 +11,7 @@
 	viable_mobtypes = list(/mob/living/carbon/human)
 	desc = "Subject is possesed by the vengeful spirit of a parrot. Call the priest."
 	severity = VIRUS_SEVERITY_MEDIUM
+	infectable_hosts = list(SPECIES_ORGANIC, SPECIES_UNDEAD, SPECIES_INORGANIC, SPECIES_ROBOTIC)
```
Comment (Supermichael777) [inline code/datums/diseases/parrotpossession.dm]: this should bypass immunity because 2spook.

**Comment by Nabski89** [comment]
I don't think the works on corpses part works.

I can't get it to show up on a scanner, or on the medhub (which makes sense). I've tried for healing, race change, and fire creating (Because lets be honest making every corpse a plasmaman corpse seems cool) and none of them worked for me.

**Comment by Nabski89** [comment]
And this should really be a bug report, but I made a virus with 

Sneezing
Necrotic Metabolism
Viral Accel
Reviltigo
Itching
Inorganic Biology

and it wouldn't process past stage one on monkeys or humans and wouldn't cure. Neutered everything but the reviltigo and it worked fine (I think).

**Pull request merged**